### PR TITLE
Add addIncludeDir to TestStep + fix build template

### LINKED ISSUE #794: builder.LibExeObjStep.addIncludeDir() not working in custom build.zig?
I'd like to call GTK functions from my zig program, here is the relevant code:
```
// build.zig
const cell = b.addExecutable("cell", ...);
cell.setBuildMode(mode);
cell.addIncludeDir("/usr/include/gtk-3.0");
```
```
// my module
const builtin = @import("builtin");
const Os = builtin.Os;
const c = switch (builtin.os) {
    Os.linux, Os.macosx => @cImport({
        @cInclude("stdio.h");
        @cInclude("stdlib.h");
        @cInclude("gtk/gtk.h");
    }),
    else => {},
}
```
But the compiler complaints about not finding the header file:
```
/home/cailei/proj/cell/infra/win.zig:6:28: error: C import failed
    Os.linux, Os.macosx => @cImport({
                           ^
/tmp/KyXYT4.h:3:10: note: 'gtk/gtk.h' file not found
#include <gtk/gtk.h>
         ^
```
And the file is apparently there:
```
$ ll /usr/include/gtk-3.0/gtk/gtk.h                                                   
-rw-r--r-- 1 root root 9663 10月 31 04:07 /usr/include/gtk-3.0/gtk/gtk.h  
```

### PR DESCRIPTION
**Pull request by tiehuis**
Closes #794.

### CODE REVIEW

File: std/build.zig [hunk 1]
```diff
@@ -1648,6 +1648,10 @@ pub const TestStep = struct {
         self.verbose = value;
     }
 
+    pub fn addIncludeDir(self: &TestStep, path: []const u8) void {
+        self.builder.include_paths.append(path) catch unreachable;
```
Comment (andrewrk) [inline std/build.zig]: This does fix the issue, but it adds the include directory globally. I think it should add the directory for only this build artifact.

**Comment by tiehuis** [comment]
It would be good to merge this before the 0.2.0 to fix the default build template else any new user would encounter an error there.

**Comment by andrewrk** [comment]
Oops, I'm so sorry. I saw this just after creating the tag.

**Comment by andrewrk** [comment]
This looks good. I'm going to add a test to `test/standalone/` and then merge.

**Comment by andrewrk** [comment]
Merged. Thanks!

I'm thinking maybe in the future we can have `--init-lib` and `--init-exe` which can be actual working examples. Then the build template can be tested in the main test suite.

**Pull request closed without merging**